src/ngx-gallery-scroll-overview/ngx-gallery-scroll-overview.component.ts

```diff
@@ -1,65 +1,62 @@
-import { Component, Input, AfterViewInit } from '@angular/core';
-import { NgxGalleryPreviewComponent } from '../ngx-gallery-preview.component';
+import { Component, Input, Output, EventEmitter, OnInit } from '@angular/core';
+import { ZoomPosition } from './zoomPosition';
 
 @Component({
     selector: 'ngx-gallery-scroll-overview',
     templateUrl: './ngx-gallery-scroll-overview.component.html',
     styleUrls: ['./ngx-gallery-scroll-overview.component.scss']
 })
-export class NgxGalleryScrollOverviewComponent implements AfterViewInit {
+export class NgxGalleryScrollOverviewComponent implements OnInit {
     private beforeZoomLeft: number;
     private beforeZoomTop: number;
     private initialLeft: number;
     private initialTop: number;
     private SCALE_FACTOR = 8;
 
-    @Input() gallery: NgxGalleryPreviewComponent;
+    @Output() onZoomChanged = new EventEmitter();
+    @Input() zoomPosition: ZoomPosition;
 
-    inBounds = true;
     edge = {
         top: true,
         bottom: true,
         left: true,
         right: true
     };
 
+    previewContainerStyles = {};
+    zoomContainerStyles = {};
+
     constructor() {
 
     }
 
     public updateDetailZoom() {
-        var topScaled = (this.initialTop - this.gallery.positionTop) / this.SCALE_FACTOR;
-        var leftScaled = (this.initialLeft - this.gallery.positionLeft) / this.SCALE_FACTOR;
+        var topScaled = (this.initialTop - this.zoomPosition.positionTop) / this.SCALE_FACTOR;
+        var leftScaled = (this.initialLeft - this.zoomPosition.positionLeft) / this.SCALE_FACTOR;
         var transformString = 'translate(' + leftScaled + 'px,' + topScaled + 'px)';
-        document.getElementById('zoomContainer').style.transform = transformString;
+        this.zoomContainerStyles["transform"] = transformString;
     }
 
     public updatePreviewScales() {
         var img = document.getElementsByClassName('ngx-gallery-fullsize');
         var width = img[0].clientWidth;
         var height = img[0].clientHeight;
 
-        // set the background image of the preview Container which contains the Scaled preview Image
-        document.getElementById('previewContainer').style.backgroundImage = "url('" + img[0].getAttribute('src') + "')";
-
         // scale the preview image
-        var widthScaled = width / this.SCALE_FACTOR;
-        var heightScaled = height / this.SCALE_FACTOR;
-        document.getElementById('previewContainer').style.width = widthScaled + 'px';
-        document.getElementById('previewContainer').style.height = heightScaled + 'px';
+        this.previewContainerStyles["background-image"] = "url('" + img[0].getAttribute('src') + "')";
+        this.previewContainerStyles["height"] = (height / this.SCALE_FACTOR) + 'px';
```

Comment:
Use dot-notation for next four lines (e.g. `this.previewContainerStyles.height`)